modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/portlet/action/GetDataMVCResourceCommand.java

```diff
@@ -423,12 +423,11 @@ locale, _fetchSegmentsExperiment(layout, segmentsExperienceId))
 	}
 
 	private String _getSegmentsExperimentActionURL(
-		String action, Group group, HttpServletRequest httpServletRequest) {
+		String action, HttpServletRequest httpServletRequest) {
 
 		return PortletURLBuilder.create(
-			_portal.getControlPanelPortletURL(
-				httpServletRequest, group,
-				SegmentsPortletKeys.SEGMENTS_EXPERIMENT, 0, 0,
+			_portletURLFactory.create(
```

Comment:
Hey @Stephy6 ,

As I understand It is better to build the URL with the requestBackedPortletURL like this  : 

               ThemeDisplay themeDisplay =
			(ThemeDisplay)httpServletRequest.getAttribute(
				WebKeys.THEME_DISPLAY);

		PortletDisplay portletDisplay = themeDisplay.getPortletDisplay();
      
                RequestBackedPortletURLFactory requestBackedPortletURLFactory =
			RequestBackedPortletURLFactoryUtil.create(httpServletRequest);

		return PortletURLBuilder.create(
			requestBackedPortletURLFactory.createActionURL(portletDisplay.getId())
		).setActionName(
			action
		).setParameter(
			"p_l_mode", Constants.VIEW
		).buildString();

Reply:
Why @Mikellorza it is better to use this approach?

Reply:
@cgoncas , as I talk with Jurgen some months ago and he told me that the url's from now on it should build with RequestBackedPortletURLFactory class.

If I am wrong let me know.

Reply:
Everytime you make a suggestion, you should know the why, what is difference between both methods and why you should recomend it to Stefan, not based just in a comment, but in facts that Stefan can understand.

Reply:
Hey @cgoncas ,

Yes you are right, but I don't know the reason a lo of things that little by little I am trying to learn. I' m just trying to contribute. Maybe it is good if you should explain thing to me if I am wrong instead attack me every time that I suggest something or something like this, because you are the person that for sure have a lo of knowledge but sometime it is difficult learn from your knowledge.

Reply:
Hey @Mikellorza !!

Thank a lot for sharing, If you are not sure about the use of this interface, could you ask it in the backend-dev channel, so other backends could explain it to us?

Thanks!!